ScoutSuite/providers/azure/resources/resources.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+
+import abc
+
+from ScoutSuite.providers.base.configs.resources import SimpleResources
+from ScoutSuite.providers.base.configs.resources import CompositeResources
+
+
+class AzureSimpleResources(SimpleResources, metaclass=abc.ABCMeta):
+
+    async def fetch_all(self):
+        raw_resource = await self.get_resources_from_api()
+        name, resource = self.parse_resource(raw_resource)
+        self[name] = resource
```

Comment:
I'm pretty confused about this method. Doesn't `get_resources_from_api` return an array of raw resources?

Reply:
It returns an array of raw resources in your case, with AWS. ;) Azure mostly returns a single value at the level of `SimpleResources`... That's why I decided to leave the classes `SimpleResources` and `CompositeResources` abstract in `base/configs/resources.py` and define `AzureSimpleResources` and `AzureCompositeResources`.